pkg/hostagent/setup.go

```diff
@@ -301,22 +301,53 @@ func (agent *HostAgent) configureContainerIfaces(metadata *md.ContainerMetadata)
 	}
 
 	podid := metadata.Id.Namespace + "/" + metadata.Id.Pod
-	{
-		agent.indexMutex.Lock()
-		if _, ok := agent.epMetadata[podid]; !ok {
-			agent.epMetadata[podid] =
-				make(map[string]*md.ContainerMetadata)
+	agent.indexMutex.Lock()
```

Comment:
Please move this section to be before the RecordMetadata. This way, we will ensure delete-before-create.